src/controllers/events.js

```diff
@@ -0,0 +1,24 @@
+const EventModel = require('../models/event');
+const { User } = require('../models/user');
+
+async function joinedVoulnteers(req, res) {
+    try {
+        const event = await EventModel.findById(req.params.id);
+        if (!event) {
+            return res.status(404).json({ message: 'Event not found' });
+        }
+        await EventModel.findByIdAndUpdate(req.params.id, {
```

Comment:
You are not checking first if the user is alreay in the array of confirmedVolunteers. This array should not contain any duplicates.